File: BinarytoDecimal.java
import java.util.Scanner;

public class BinarytoDecimal {
    public static int BinarytoDecimal(Long binaryNumber) {
        String value = Long.toString(binaryNumber);
        int length = value.length();
        int output = 0;
        for (int position = 0; position < length; position++) {
            char chr = value.charAt(position);
            if (chr == '1') {
                output += Math.pow(2, length - 1 - position);
            } else if (chr == '0') {
                continue;
            } else {
                System.err.println("Make sure you only input binary values");
                System.exit(1);
            }
        }
        return output;
    }
    public static void main(String[] args) {
        System.out.print("Binary Number: ");
        Scanner inp = new Scanner(System.in);
        System.out.println(BinarytoDecimal(inp.nextLong()));
        inp.close();
    }
}

File: BinarytoDecimal.c
/* gcc BinarytoDecimal.c -lm */
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int BinarytoDecimal(int binaryNumber) {
    if (binaryNumber == 0) {
        return 0;
    }
    int len = floor(log10(binaryNumber)) + 1;
    char value[len + 1];
    sprintf(value, "%d", binaryNumber);
    int output = 0;
    for (int position = 0; position < len; position++) {
        char chr = value[position];
        if (chr == '1') {
            output += pow(2, len - 1 - position);
        } else if (chr == '0') {
	    continue;
        } else {
            printf("%s\n", "Make sure you only input binary values");
	    exit(-1);
        }
    }
    return output;
}

int main(void) {
    int num;
    printf("%s", "Binary Number: ");
    scanf("%d", &num);
    printf("%d\n", BinarytoDecimal(num));
}
